src/components/tree/phyloTree/helpers.js

```diff
@@ -99,6 +103,51 @@ export const setDisplayOrder = (nodes) => {
   nodes[0].displayOrderRange = [undefined, undefined];
 };
 
+/**
+ * @param {Array<PhyloNode>} nodes
+ * @param {boolean} focus
+ * @returns fn to return a display order (y position) for a node
+ */
+function getDisplayOrderCallback(nodes, focus) {
+  /**
+   * Start at 0 and increase with each node.
+   * Note that this value is shared across invocations of the callback.
+   */
+  let displayOrder = 0;
+
+  /**
+   * Keep track of whether the previous node was selected
+   */
+  let previousWasVisible;
+
+  if (focus) {
+    const numVisible = nodes.filter((d) => !d.hasChildren && d.visibility === NODE_VISIBLE).length;
+    const yProportionFocused = Math.max(0.8, numVisible / nodes.length);
```

Comment:
Mentioned above, and not a big deal for the v1 of this feature, but when we have subtrees this no longer represents the proportion of vertical space the focused tips will take up as the space between subtrees hasn't been accounted for. The more subtrees there are the more this inter-subtree space will start to dominate.

Reply:
With 512b530980b89dcf37749adcedbd1ae2548d789a there is a slight difference in behavior when all tips are selected and toggling focus, but I'm thinking overall it's better.

Reply:
Looks good - you're right about the slight difference in behaviour, but this highlights a subtle issue where we're confusing `nodes.length` with total number of tips. Fixing this improves other things as well!

```diff
- const yPerFocused = (yProportionFocused * nodes.length) / numVisible;
- const yPerUnfocused = ((1 - yProportionFocused) * nodes.length) / (nodes.length - numVisible);
+ const yPerFocused = (yProportionFocused * nodes[0].n.fullTipCount) / numVisible;
+ const yPerUnfocused = ((1 - yProportionFocused) * nodes[0].n.fullTipCount) / (nodes[0].n.fullTipCount - numVisible);
```

Reply:
Good catch! Fixed and used a new variable for clarity: https://github.com/nextstrain/auspice/blob/126b895d6cc0623b6a1b5534873291f6d8977bd3/src/components/tree/phyloTree/helpers.js#L102-L112